narwhal/primary/src/core.rs

```diff
@@ -180,6 +179,51 @@ impl Core {
 
         self.highest_received_round = last_round_number;
         self.highest_processed_round = last_round_number;
+
+        // Get latest header from all peers and process it.
+        let peers: Vec<NetworkPublicKey> = self
+            .committee
+            .others_primaries(&self.name)
+            .into_iter()
+            .map(|(_, _, network_key)| network_key)
+            .collect();
+
+        let network = P2pNetwork::new(self.network.network());
```

Comment:
`P2pNetwork::New` constructs new rng, bounded executor map, etc, which you don't actually need since none of it is used with the method you're calling. can you use the anemo::Network type directly below instead of passing it through an extra `P2pNetwork` (which just gets removed again when you call `network.network()` a second time on line 195)

alternatively, instead of adding your new method to `PrimaryToPrimaryRpc`, you could `impl UnreliableNetwork<LatestHeaderRequest> for P2pNetwork` and then call `network.unreliable_broadcast(request)`? that would let you avoid duplicating the code for looking up all the other peers.

Reply:
I made the first suggested update. If we use `impl UnreliableNetwork<LatestHeaderRequest> for P2pNetwork` the trait must return an empty result, whereas we need one that contains the header.

Reply:
I don't think that's true? the return type of `unreliable_broadcast` is Vec<Result<JoinHandle<Result<anemo::Response<Self::Response>>>>>

So you'd get back all of the join handles for the broadcasted requests on which you could await to get the response